### PR TITLE
Refactoring for `ImmersedBoundaries` to take precedence over `Solvers`

### PR DESCRIPTION
**Pull request by simone-silvestri**
We discussed this for a while and decided that this makes sense. Now, there are also some active use cases to justify pursuing this.

This PR is an attempt to do this. Let's see how much code breaks...

cc @ali-ramadhan @tomchor @glwagner

### CODE REVIEW

**Comment by ali-ramadhan** [comment]
This PR should also resolve https://github.com/CliMA/Oceananigans.jl/issues/3829!

File: src/ImmersedBoundaries/active_cells_map.jl [hunk 1]
```diff
@@ -24,13 +19,10 @@ const WholeActiveCellsMapIBG = ImmersedBoundaryGrid{<:Any, <:Any, <:Any, <:Any,
...
 const SplitActiveCellsMapIBG = ImmersedBoundaryGrid{<:Any, <:Any, <:Any, <:Any, <:Any, <:Any, <:NamedTuple}
 
-# A distributed grid with split interior map
-const DistributedActiveCellsIBG = ImmersedBoundaryGrid{<:Any, <:Any, <:Any, <:Any, <:DistributedGrid, <:Any, <:NamedTuple} 
-
 """
 A constant representing an immersed boundary grid, where interior active cells are mapped to linear indices in grid.interior_active_cells
 """
-const ActiveCellsIBG = Union{DistributedActiveCellsIBG, WholeActiveCellsMapIBG, SplitActiveCellsMapIBG}
+const ActiveCellsIBG = ImmersedBoundaryGrid{<:Any, <:Any, <:Any, <:Any, <:Any, <:Any, <:Union{AbstractArray, NamedTuple}}
```
Comment (glwagner) [inline src/ImmersedBoundaries/active_cells_map.jl:25]: this is unrelated right

File: src/Coriolis/hydrostatic_spherical_coriolis.jl [hunk 2]
```diff
@@ -49,6 +50,7 @@ Adapt.adapt_structure(to, coriolis::HydrostaticSphericalCoriolis) =
 
 @inline φᶠᶠᵃ(i, j, k, grid::LatitudeLongitudeGrid)        = φnode(j, grid, Face())
 @inline φᶠᶠᵃ(i, j, k, grid::OrthogonalSphericalShellGrid) = φnode(i, j, grid, Face(), Face())
+@inline φᶠᶠᵃ(i, j, k, grid::ImmersedBoundaryGrid)         =  φᶠᶠᵃ(i, j, k, grid.underlying_grid)
```
Comment (glwagner) [inline src/Coriolis/hydrostatic_spherical_coriolis.jl:53]:
```suggestion
@inline φᶠᶠᵃ(i, j, k, grid::ImmersedBoundaryGrid)         = φᶠᶠᵃ(i, j, k, grid.underlying_grid)
```

File: src/ImmersedBoundaries/active_cells_map.jl (hunk 1 above)
Comment (simone-silvestri) [reply src/ImmersedBoundaries/active_cells_map.jl:25]: this had to change because `DistributedActiveCellsIBG` is now defined after the `ImmersedBoundaries` module
